pkg/watch/manager.go

```diff
@@ -234,7 +234,7 @@ func (wm *Manager) doRemoveWatch(r *Registrar, gvk schema.GroupVersionKind) erro
 }
 
 // replaceWatches ensures all and only desired watches are running.
-func (wm *Manager) replaceWatches(r *Registrar) error {
+func (wm *Manager) replaceWatches(ctx context.Context, r *Registrar) error {
```

Comment:
What would the desired behavior here be if we had a context that got canceled?

Reply:
Now we're passing it as opencensus asks for context when reporting metrics.

Reply:
Just so I understand, OpenCensus uses context for tag propagation, not cancelation, correct? Where do these extra tags come from? Is the use case distributed tracing? Is it relevant for metrics?

Reply:
Correct, just tag propagation. If the tag isn't registered in the Context, it won't be recorded

Reply:
Per Max's suggestion, I've made contexts instantiated per-call for this case rather than having to be passed the entire call stack

Reply:
Done.